src/chainstate/stacks/mod.rs

```diff
@@ -20,69 +20,192 @@
+            Error::InvalidStacksTransaction(ref s) => fmt::Display::fmt(s, f),
+            Error::PostConditionFailed(ref s) => fmt::Display::fmt(s, f),
+            Error::NoSuchBlockError => f.write_str(error::Error::description(self)),
+            Error::InvalidChainstateDB => f.write_str(error::Error::description(self)),
+            Error::BlockTooBigError => f.write_str(error::Error::description(self)),
+            Error::MicroblockStreamTooLongError => f.write_str(error::Error::description(self)),
+            Error::ClarityError(ref e) => fmt::Display::fmt(e, f),
+            Error::ClarityInterpreterError(ref e) => f.write_str(&format!("{:?}", e)),
+            Error::DBError(ref e) => fmt::Display::fmt(e, f),
+            Error::NetError(ref e) => fmt::Display::fmt(e, f),
+            Error::MARFError(ref e) => fmt::Display::fmt(e, f),
         }
     }
 }
 
 impl error::Error for Error {
     fn cause(&self) -> Option<&dyn error::Error> {
         match *self {
-            Error::EncodeError => None,
-            Error::DecodeError => None,
-            Error::AuthError => None,
+            Error::InvalidFee => None,
+            Error::InvalidStacksBlock(ref _s) => None,
+            Error::InvalidStacksMicroblock(ref _s, ref _h) => None,
+            Error::InvalidStacksTransaction(ref _s) => None,
+            Error::PostConditionFailed(ref _s) => None,
+            Error::NoSuchBlockError => None,
+            Error::InvalidChainstateDB => None,
+            Error::BlockTooBigError => None,
+            Error::MicroblockStreamTooLongError => None,
+            Error::ClarityError(ref e) => Some(e),
+            Error::ClarityInterpreterError(ref e) => None,
+            Error::DBError(ref e) => Some(e),
+            Error::NetError(ref e) => Some(e),
+            Error::MARFError(ref e) => Some(e),
         }
     }
 
     fn description(&self) -> &str {
         match *self {
-            Error::EncodeError => "Failed to encode",
-            Error::DecodeError => "Failed to decode",
-            Error::AuthError => "Failed to authenticate transaction",
+            Error::InvalidFee => "Invalid fee",
+            Error::InvalidStacksBlock(ref s) => s.as_str(),
+            Error::InvalidStacksMicroblock(ref s, ref h) => s.as_str(),
+            Error::InvalidStacksTransaction(ref s) => s.as_str(),
+            Error::PostConditionFailed(ref s) => s.as_str(),
+            Error::NoSuchBlockError => "No such Stacks block",
+            Error::InvalidChainstateDB => "Invalid chainstate database",
+            Error::BlockTooBigError => "Too much data in block",
+            Error::MicroblockStreamTooLongError => "Too many microblocks in stream",
+            Error::ClarityError(ref e) => e.description(),
+            Error::ClarityInterpreterError(ref e) => "Clarity fucked up and Aaron didn't make this error struct implement the Error trait, so who knows what went wrong?",
```

Comment:
👀

Reply:
Lol that's not going to make it to `master`!

More seriously, I think this right here was also the reason I modified the code in `src/vm/clarity.rs` to use `Error` instead of `CheckError` (which you commented on earlier).

Reply:
lol, but both implement `err::Error`!

I think there's just not much useful information provided by that trait (e.g., you don't get anything at all like a stack trace, and .source is only useful for wrapped errors) -- check errors however _do_ try to provide more information via the `DiagnosableError` trait, and the interpreter runtime errors also try to provide Clarity stack information.

Reply:
should be fixed now -- this error is removed